File: javascript/src/models/googleCloudPubSubConfig.ts
// this file is @generated

export interface GoogleCloudPubSubConfig {
  projectId: string;
  topicId: string;
  /** Google Cloud Credentials JSON Object as a string. */
  credentials: string;
}

export const GoogleCloudPubSubConfigSerializer = {
  _fromJsonObject(object: any): GoogleCloudPubSubConfig {
    return {
      projectId: object["projectId"],
      topicId: object["topicId"],
      credentials: object["credentials"],
    };
  },

  _toJsonObject(self: GoogleCloudPubSubConfig): any {
    return {
      projectId: self.projectId,
      topicId: self.topicId,
      credentials: self.credentials,
    };
  },
};

File: kotlin/lib/src/main/kotlin/models/GoogleCloudPubSubConfig.kt
// This file is @generated
package com.svix.kotlin.models

import kotlinx.serialization.Serializable

@Serializable
data class GoogleCloudPubSubConfig(
    val projectId: String,
    val topicId: String,
    /** Google Cloud Credentials JSON Object as a string. */
    val credentials: String,
)
